### ISSUE TITLE
[Question] Optimizing binary array

### ISSUE THREAD

**Issue by astanziola** [RobertTLange/evosax#66]
First of all, thanks for the amazing library 😄 

What's the best way to used it to optimize a binary array? The simplest way I can think about is still to optimize a continuous array, clipped between 0 and 1, and apply tresholding in the fitness function, but perhaps there's a way to directly use boolean arrays?

Thanks!

**Comment by maxencefaldor** [comment]
Yes, optimize a continuous array and applying thresholding is a potential solution.

What I would recommend to do is to optimize an continuous array that represents logits modelling the probability of each element in the array to be 0 or 1 (i.e., Bernoulli distribution).

For example, the array `a` could take values between `-jnp.inf` and `jnp.inf` and to sample candidate solutions from it, you would need to first apply sigmoid to transform the logits into probabilities, and then sample according to the probabilities with
```
binary_array = jax.random.uniform(a.shape) < jax.nn.sigmoid(a)
```

I am closing the issue for now. Don't hesitate to reopen it if you have any questions.
